src/ess.jl

```diff
@@ -144,16 +144,7 @@ function mean_autocov(k::Int, cache::ESSCache)
     firstrange = 1:(niter - k)
     lastrange = (k + 1):niter
     s = Statistics.mean(1:nchains) do i
-        if eltype(samples) isa LinearAlgebra.BlasReal
```

Comment:
Yes, this is incorrect and should be fixed. I think it would be good to benchmark the corrected version, in particular also with older Julia versions in which AFAIK `view`s were more expensive. The special `dot` call without views is e.g. also used in StatsBase: https://github.com/JuliaStats/StatsBase.jl/blob/2faa6e80b7966b915086d8cd5a4a4d89a2126db5/src/signalcorr.jl#L46 (where it was copied from I assume).

Reply:
> The special `dot` call without views is e.g. also used in StatsBase: https://github.com/JuliaStats/StatsBase.jl/blob/2faa6e80b7966b915086d8cd5a4a4d89a2126db5/src/signalcorr.jl#L46 (where it was copied from I assume).

It's not quite the same. In StatsBase, an `AbstractVector` is passed to `dot` This versions passes an `AbstractMatrix`, but it errors because there is no corresponding method. I don't think this branch ever worked, and I'm not certain how best to correct it for benchmarking.

Reply:
Ah I missed that it also incorrectly used `samples` instead of `view(samples, :, i)`. But in this case one already creates a view 🤷🏻‍♂️ Of course, one could linearize the matrix with `vec` and update the ranges in every step but it seems unnecessarily complicated. Let's just drop the branch, since it hasn't been used there can't be any regressions anyway 👍